junit-runtime/src/main/scala/org/hamcrest/internal/ArrayIterator.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Ported from https://github.com/hamcrest/JavaHamcrest/
+ */
+package org.hamcrest.internal
+
+import java.{util => ju}
+
+class ArrayIterator(array: Array[AnyRef], private var currentIndex: Int = 0)
```

Comment:
This constructor does not appear to exist in the origin library (https://github.com/hamcrest/JavaHamcrest/blob/master/hamcrest-core/src/main/java/org/hamcrest/internal/ArrayIterator.java). I suggest making the other one the primary constructor and removing this one.